### PR TITLE
Add support for Clipsal cent-a-meter CMR113 power meter

### PR DESCRIPTION
**Pull request by mikey**
Device only measures current but is only sold in Australia so driver
gives power output assuming voltage is 240V.

### CODE REVIEW

File: src/devices/cmr113.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	    reflect_bytes(buf, 2);
+	    current += ((float)buf[0] + ((buf[1] & 0x3) << 8))/10.0;
+    }
+    /* These seem to be exclusively in Australia, so assume 240V */
+    power = 240 * current;
+
+    data = data_make(
+	    "model",   "",         DATA_STRING, "CMR112",
+	    "power_W", "Power",    DATA_FORMAT, "%.0f W", DATA_DOUBLE,  power,
+	    "current", "Current",  DATA_FORMAT, "%.1f A", DATA_DOUBLE,  current,
```
Comment (merbanan) [inline src/devices/cmr113.c:81]: SI unit missing.

Comment (merbanan) [inline src/devices/cmr113.c:81]: SI unit missing.

File: examples/rtl_433_mqtt_relay.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 UDP_IP = "127.0.0.1"
 UDP_PORT = 1433
 
-MQTT_HOST = "127.0.0.1"
+MQTT_HOST = "192.168.86.37"
 MQTT_PORT = 1883
```
Comment (merbanan) [inline examples/rtl_433_mqtt_relay.py:22]: Stray hunk

**Comment by merbanan** [comment]
Can you add a bit stream description also? And then we need some sample signals also.

File: examples/rtl_433_mqtt_relay.py (hunk 2 above)
Comment (mikey) [reply examples/rtl_433_mqtt_relay.py:22]: oops, yes. Sorry

**Comment by mikey** [comment]
> Can you add a bit stream description also? And then we need some sample signals also.

Yes will do. I have some good samples. I just need to take some photos of the device.

**Comment by merbanan** [comment]
Not sure we should output the power. The power depends on the voltage. In regular grids the voltage can vary up to 5-10% and still be ok. But the actual power will then change depending on that. So I prefer to only relay data that is actually coming from the transmitter.

File: src/devices/cmr113.c (hunk 1 above)
Comment (mikey) [reply src/devices/cmr113.c:81]: SI missing on the current? should be "current_A" ?

Comment (merbanan) [reply src/devices/cmr113.c:81]: I think that is the correct unit.

**Comment by mikey** [comment]
> Not sure we should output the power. The power depends on the voltage. In regular grids the voltage can vary up to 5-10% and still be ok. But the actual power will then change depending on that. So I prefer to only relay data that is actually coming from the transmitter.

Fair enough. It would be nice to be able to add data about your configuration so we could do this. 

I'm also using the TS-FT002. Being able to add information about your water tank would enable us to broadcast the tank %. 

But maybe that's out of scope for this (already awesome) project :-)

**Comment by mikey** [comment]
> Not sure we should output the power. The power depends on the voltage. In regular grids the voltage can vary up to 5-10% and still be ok. But the actual power will then change depending on that. So I prefer to only relay data that is actually coming from the transmitter.

Would you also like me to keep the three current sensors separate, rather than aggregating them?

**Comment by zuckschwerdt** [comment]
Convert tabs to 4 spaces, model should likely be `Clipsal-CMR112`, run `maintainer_update.py` to get e.g. example conf right.

**Comment by merbanan** [comment]
> Would you also like me to keep the three current sensors separate, rather than aggregating them into one?

Not sure I follow you. Each transmitters data should be translated to a message. As direct translation as possible.

**Comment by mikey** [comment]
> > Would you also like me to keep the three current sensors separate, rather than aggregating them into one?
> 
> Not sure I follow you. Each transmitters data should be translated to a message. As direct translation as possible.

OK. There is only 1 transmitter but it contains data for the 3 attached current sensors in each transmission. 

I'll output 3 separate current values (I'm just output the sum of the 3 currently).

**Comment by zuckschwerdt** [comment]
> I'll output 3 separate current values (I'm just output the sum of the 3 currently).

Preferred keys then are: `current_1_A`, ...

**Comment by mikey** [comment]
Pushed some updates:
1. 4 space tabs
2. Ran maintainer_update.py (BTW SoapySDR needs some updates according to it)
3. Remove stray hunk
4. Updated bitstream description 
5. Added SI unit to output
6. Removed power output and split current in 3 (current called current_1_A, current_3_A, current_3_A)
7. Updated model number

**Comment by mikey** [comment]
@merbanan I've added some samples in https://github.com/merbanan/rtl_433_tests/pull/401

**Comment by zuckschwerdt** [comment]
> samples in [merbanan/rtl_433_tests#401](https://github.com/merbanan/rtl_433_tests/pull/401)

Can you add back the freq/rate info to the filenames, it's the only metadata we have.

**Comment by zuckschwerdt** [comment]
The `if` branch L55 needs more indent. (Clang warns about those things). `){` needs a space.
Indent on continued line, e.g. 97 107 117 is 8 spaces. (Try clang-format if you have Clang, it *mostly* gets things right.)

**Comment by mikey** [comment]
> The `if` branch L55 needs more indent. (Clang warns about those things). `){` needs a space.
I understand these two and have fixed

> Indent on continued line, e.g. 97 107 117 is 8 spaces. (Try clang-format if you have Clang, it _mostly_ gets things right.)

I think I've fixed this.

**Comment by zuckschwerdt** [comment]
Looks good, one thing I just notice: the "cmr112" should all be "cmr113", right?

**Comment by mikey** [comment]
yes, sorry. I've updated the PR title now. I can't update the branch name without closing this PR. 

The code is correct and only refers to cmr113.

**Comment by mikey** [comment]
> Branch name isn't an issue. But check the "Files changed" tab for the contents of the PR, there's 112 in multiple places.

Sorry, you're right. My bad, I 'git grepped' badly. Fixed now.

**Comment by mikey** [comment]
I've updated the tests as well as that had the same mistake.

**Comment by zuckschwerdt** [comment]
No worries. There is still one instance in example conf.
Also "model" needs to be "Clipsal-CMR113" (note the dash),
and output_fields needs to have: "model", "current_1_A", "current_2_A", "current_3_A", NULL.

**Comment by mikey** [comment]
> No worries. There is still one instance in example conf.
:facepalm: Fixed

> Also "model" needs to be "Clipsal-CMR113" (note the dash),

OK

> and output_fields needs to have: "model", "current_1_A", "current_2_A", "current_3_A", NULL.

OK

I've repushed and also fixed up the tests to add the dash. Sorry for the noise.

**Pull request merged**